js/cardview.js

```diff
@@ -16,28 +16,120 @@ let Card = React.createClass({
+    if (toIndex != -1 && hasActiveCard) {
+      this.attack(fromAttackIndex, toIndex, attackingCard)
+      return;      
+    }
+
+    // PLAY CARD FROM HAND
+    // can click to play cards in hand
+    var fromIndex = window.game.current().hand.indexOf(this.props.cardInfo);
+    if (fromIndex != -1) {
+      this.playFromHand(fromIndex, this.props.cardInfo);
       return;
     }
     
-    this.props.hasFocus = !this.props.hasFocus;
-    // play card on 2nd click
-    if (!this.props.hasFocus) {
-      var fromIndex = window.game.current().hand.indexOf(this.props.cardInfo);
-      console.log(fromIndex);
-      console.log(window.game.current().hand);
-      var move = {"op":"play", 
+    // HIGHLIGHT CARD, OR SMASH FACE IF ALREADY HIIGHLIGHTED
+    // can click a card in play if it's in the player's board
+    // and it's not summoning sick or already attacked
+    fromIndex = window.game.current().board.indexOf(this.props.cardInfo);
+    if (fromIndex != -1 && 
+        !this.props.cardInfo.enteredPlayThisTurn &&
+        !this.props.cardInfo.hasAttacked) {
+      this.clickCardInPlay(fromIndex, this.props.cardInfo);
+      return;
+    }
+
+    // if it's not in their hand or board, then do nothing
+
+  },
+  
+  // smash creature's face
+  attackCreature: function(fromAttackIndex, toIndex, attackingCard) {
+    var move = {"op":"attack", 
+                "from":fromAttackIndex,
+                "to": toIndex
+               };
+    window.client.makeLocalMove(move);
+    window.client.gameView.forceUpdate();
+    attackingCard.hasAttacked = true;
+  },
+
+  // highlight a card when clicked, play when double clicked
+  playFromHand: function(fromIndex, card) {
+    // card can only be highlighted and played if player has enough mana
+    if (this.props.cardInfo.cost < this.props.player.mana) {
+      let moveClosure = function() {
+        var move = {"op":"play", 
+                    "from":fromIndex
+                   };
+        window.client.makeLocalMove(move);
+        card.enteredPlayThisTurn = true;
+      }
+      this.highlightOrPlayMove(moveClosure);    
+    }
+  },
+
+  // highlight a card when clicked, attack face when double clicked
+  clickCardInPlay: function(fromIndex, card) {
+    let moveClosure = function() {
+      var move = {"op":"face", 
                   "from":fromIndex
                  };
       window.client.makeLocalMove(move);
-    } else {
-      this.forceUpdate();
+      card.hasAttacked = true;      
     }
+    this.highlightOrPlayMove(moveClosure);
   },
 
+  // when a card is clicked, highlight it, play it, or attack with it
+  highlightOrPlayMove: function(moveClosure) {
+    this.props.cardInfo.hasFocus = !this.props.cardInfo.hasFocus;
+    if (!this.props.cardInfo.hasFocus) { // play or attack with card
+      moveClosure();
+      window.client.gameView.forceUpdate();
+    } else { // just highlight the card
+      this.forceUpdate();
```

Comment:
I think we need to add the first tap of a card to activate it, into the
game state?

Andrew Johnson
Founder
*gaiagps.com <http://gaiagps.com>*

On Sat, Jun 13, 2015 at 7:04 AM, Kevin Lacker <notifications@github.com>
wrote:

> In js/cardview.js
> <https://github.com/lacker/cardkit/pull/14#discussion_r32370707>:
>
> >    },
> >
> > +  // when a card is clicked, highlight it, play it, or attack with it
> > +  highlightOrPlayMove: function(moveClosure) {
> > +    this.props.cardInfo.hasFocus = !this.props.cardInfo.hasFocus;
> > +    if (!this.props.cardInfo.hasFocus) { // play or attack with card
> > +      moveClosure();
> > +      window.client.gameView.forceUpdate();
> > +    } else { // just highlight the card
> > +      this.forceUpdate();
>
> ditto, why isn't makeLocalMove doing this already.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lacker/cardkit/pull/14/files#r32370707>.
>